corehq/apps/geospatial/models.py

```diff
@@ -93,8 +93,13 @@ def plaintext_api_token(self):
 
     @plaintext_api_token.setter
     def plaintext_api_token(self, value):
-        if value and not value.startswith(f'${ALGO_AES}$'):
-            ciphertext = b64_aes_encrypt(value)
-            self.api_token = f'${ALGO_AES}${ciphertext}'
-        else:
-            self.api_token = None
+        try:
+            if value is None:
+                self.api_token = None
+            elif value and not value.startswith(f'${ALGO_AES}$'):
+                ciphertext = b64_aes_encrypt(value)
+                self.api_token = f'${ALGO_AES}${ciphertext}'
+            else:
+                raise Exception("Unexpected value set for plaintext api token")
+        except AttributeError:
```

Comment:
Thanks for the improvement @Charl1996 
Can you share what will raise the exception `AttributeError`?

Additionally, possible to keep the two messages different? If one gets an error, it wont be clear where it came from.

Reply:
When raising exceptions its nice to be explicit what code would raise the exception. It's not clear to me here what will raise the exception `AttributeError`, though if only the necessary code is kept in the try block, its easier to locate.